packages/providers/src.ts/base-provider.ts

```diff
@@ -1256,21 +1256,35 @@ export class BaseProvider extends Provider implements EnsProvider {
         }
     }
 
-    async getCode(addressOrName: string | Promise<string>, blockTag?: BlockTag | Promise<BlockTag>): Promise<string> {
+    /**
+     *  Get contract bytecode implementation, using the REST Api.
+     *  It returns the bytecode, or a default value as string.
+     *
+     * @param addressOrName The address to obtain the bytecode of
+     */
+    async getCode(addressOrName: string | Promise<string>, throwOnNonExisting?: boolean): Promise<string> {
         await this.getNetwork();
-        const params = await resolveProperties({
-            address: this._getAddress(addressOrName),
-            blockTag: this._getBlockTag(blockTag)
-        });
-
-        const result = await this.perform("getCode", params);
+        addressOrName = await addressOrName;
+        const { shard, realm, num } = getAccountFromAddress(addressOrName);
+        const shardNum = BigNumber.from(shard).toNumber();
+        const realmNum = BigNumber.from(realm).toNumber();
+        const accountNum = BigNumber.from(num).toNumber();
+        const endpoint = '/api/v1/contracts/' + shardNum + '.' + realmNum + '.' + accountNum;
         try {
-            return hexlify(result);
+            let { data } = await axios.get(this.mirrorNodeUrl + endpoint);
+            if (data.bytecode != null) {
+                return hexlify(data.bytecode);
+            }
+            return "0x";
         } catch (error) {
-            return logger.throwError("bad result from backend", Logger.errors.SERVER_ERROR, {
-                method: "getCode",
-                params, result, error
-            });
+            if (throwOnNonExisting) {
```

Comment:
what if the error is not due to non existing contract? Maybe the API returns 500 or 403 or something other than 404. In that case we should not return `0x`